### ISSUE TITLE
Spork doesn't seem to run SimpleCov correctly with RSpec and Rails

### ISSUE THREAD

**Issue by fj** [colszowka/simplecov#42]
**tl;dr:**

Summary of results:

<table>
<tr><th rowspan="2" colspan="2">summary of issues</th><th colspan="2">Spork</th></tr>
<tr><th>not running</th><th>running</th></tr>
<tr><th rowspan="2">SimpleCov</th><th>top of Spork.prefork</th><td>works as expected</td><td>SimpleCov doesn't run</td></tr>
<tr><th>top of Spork.each_run</th><td>always 0 LOC, 0% coverage</td><td>SimpleCov doesn't run</td></tr>
</table>

<hr/>

I'm on an existing Rails project using Mongoid, Devise, RSpec, and Spork (0.9.0.rc5). Our `spec/spec_helper.rb` looks like this:

```ruby
require 'spork'

ENV["RAILS_ENV"] ||= 'test'

Spork.prefork do
  require 'simplecov'
  SimpleCov.start 'rails'

  # trap mongoid
  require 'rails/mongoid'
  Spork.trap_class_method(Rails::Mongoid, :load_models)

  # trap devise
  require 'rails/application'
  Spork.trap_method(Rails::Application, :reload_routes!)

  require File.expand_path("../../config/environment", __FILE__)
  require 'rspec/rails'
  Dir[Rails.root.join("spec/support/**/*.rb")].each {|f| require f}

  DatabaseCleaner.strategy = :truncation

  RSpec.configure do |config|
    config.mock_with :rspec

    config.before(:each) do
      DatabaseCleaner.clean
    end
  end
end

Spork.each_run do
  # reload factories
  Factory.definition_file_paths = [
    File.join(Rails.root, 'spec', 'factories')
  ]
  Factory.find_definitions
end
```

This configuration will run SimpleCov when Spork is _not_ running, and it produces the expected result. When Spork _is_ running, no coverage message is shown.

If I move the `require 'simplecov'; SimpleCov.start 'rails'` lines to the beginning of the Spork.each_run block, as SimpleCov's readme suggests, then when Spork is not running, I get this:

```
Coverage report generated for RSpec to /home/johnf/projects/<elided>/coverage. 0 / 0 LOC (0.0%) covered.
```

If Spork _is_ running, then instead I get no coverage message, as before.

**Comment by bf4** [comment]
ref: https://github.com/colszowka/simplecov/issues/340

**Comment by bf4** [comment]
Closing as stale
